xml-conduit/src/Text/XML/Stream/Parse.hs

```diff
@@ -659,13 +659,21 @@ parseContent (ParseSettings decodeEntities _ decodeIllegalCharacters _) breakDou
     case toValidXmlChar n <|> decodeIllegalCharacters n of
       Nothing -> fail "Invalid character from decimal character reference."
       Just c  -> return $ ContentText $ T.singleton c
-  parseTextContent = ContentText <$> takeWhile1 valid <?> "text content"
+  parseTextContent = ContentText . normalizeLineEndings <$> takeWhile1 valid <?> "text content"
   valid '"'  = not breakDouble
   valid '\'' = not breakSingle
   valid '&'  = False -- amp
   valid '<'  = False -- lt
   valid _    = True
 
+-- | Turns @\r\n@ and @\r@ into @\n@. See
+-- <https://www.w3.org/TR/REC-xml/#sec-line-ends>.
+normalizeLineEndings :: Text -> Text
+normalizeLineEndings =
+  -- NOTE: The order of these 'T.replace' calls is important.
+  -- If it was reversed, @\r\n@ would become @\n\n@.
+  T.replace "\r" "\n" . T.replace "\r\n" "\r"
```

Comment:
Would it be more efficient to implement this replacement at the `attoparsec` parser level ? I'm worried that the current code could cause all XML content to be traversed twice (once for `takewhile1`, and once for both `T.replace`, assuming they are subject to fusion), but I'm not sure.